Jenkinsfile

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env groovy
+
+stage ('Parallel Build') {
+
+    def variants = [
+        'clang.debug.unity',
+        'clang.debug.nounity',
+        'gcc.debug.unity',
+        'gcc.debug.nounity',
+        'clang.release.unity',
+        'gcc.release.unity',
+        'coverage'] as String[]
+
+    // create a map of all builds
+    // that we want to run. The map
+    // is string keys and node{} object values
+    def builds = [:]
+    for (int index = 0; index < variants.size(); index++) {
+        def bldtype = variants[index]
+        builds[bldtype] = {
+            node('rippled-dev') {
+                checkout scm
+                dir ('build') {
+                    deleteDir()
+                }
+                def cdir = upDir(pwd())
+                echo "BASEDIR: ${cdir}"
+                def compiler = getCompiler(bldtype)
+                def target = getTarget(bldtype)
+                if (compiler == "coverage") {
+                    compiler = 'gcc'
+                }
+                echo "COMPILER: ${compiler}"
+                echo "TARGET: ${target}"
+                def clang_cc  = (compiler == "clang") ? '/opt/llvm-4.0.0/bin/clang' : ''
+                def clang_cxx = (compiler == "clang") ? '/opt/llvm-4.0.0/bin/clang++' : ''
```

Comment:
right - llvm was installed using these packages:  https://copr-be.cloud.fedoraproject.org/results/alonid/llvm-4.0.0/epel-7-x86_64/  which are conveniently dropping them into `/opt/llvm-VERSION`. Yeah, I can absolutely make and LLVM_DIR var and reference that here. This will be part of the slave env settings (configured in jenkins) - we already define BOOST_ROOT and a few others this way.